src/smt/set_defaults.cpp

```diff
@@ -314,6 +319,16 @@ void SetDefaults::setDefaultsPre(Options& opts)
             smt, proofMode, options::ProofMode::PP_ONLY, "produce difficulty");
       }
     }
+    if (opts.proof.proofLog)
+    {
+      SET_AND_NOTIFY(smt, produceProofs, true, "proof logging");
+      // ensure at least preprocessing proofs are enabled
```

Comment:
why this?

Reply:
This ensures we have initialized a proof node manager at least.  I suppose it is possible to make an even lower `sat-proof` mode where our proof infrastructure is initialized but nothing is told to produce proofs, do you think we should add this?

Reply:
what would the proof node look like in that case? We could have something like what we have for SAT_REFUTATION, but each preprocessed input is justified directly by PREPROCESSED_LEMMA or something.

Reply:
Yes, what you are suggesting is implemented by this PR (see implementation of `ProofLoggerCpc::logCnfPreprocessInputs` and the trust id added on this branch.

However, indeed this method is never called, since we always assume that if a proof manager exists, then preprocessing proofs are enabled. 

We can do this but perhaps in another PR, I can open a wishue to track this.

Reply:
Agreed, thanks!